### DISCUSSION TITLE
How to highlight the code block

### DISCUSSION THREAD

**Discussion by ZacharyBear** [remarkjs/remark#1227] [Q&A]
I wrote a code block similar, remark just put it into a `pre>code` html structure.
The keywords are not highlight, how could I make it colorful like by using highlight.js or prismjs?

**Comment by wooorm** [comment]
You can use highlight.js or prism.
I’d recommend https://github.com/rehypejs/rehype-highlight

**Reply by ZacharyBear** [reply]
excusme, how to integrate `remark` and `rehype-highlight`?🙏

**Reply by wooorm** [reply]
We’ve got a lot of docs! You should read some. I recommend starting with https://github.com/unifiedjs/unified. Then the remark and rehype readmes.

Then, you specifically need https://github.com/remarkjs/remark-rehype

**Reply by ZacharyBear** [reply]
thanks, I'll follo this way

**Comment by ZacharyBear** [comment]
![image](https://github.com/remarkjs/remark/assets/62818957/b109c2d7-5851-4e4d-8187-bab35f60edb8)
I did the highlight through the code fragement below:
```js
  const file = await unified()
    .use(remarkParse)
    .use(remarkRehype)
    .use(rehypeDocument, {title: '👋🌍'})
    .use(rehypeFormat)
    .use(rehypeStringify)
    .use(rehypeHighlight)
    .process(await read(path.join('src/posts', 'test.md')))
```
It:
1. Parse Markdown by `remark-parse`
2. Turn markdown to HTML by `remark-rehype`
3. Add a *<title>* tag to the HTML content through `remark-document`
4. Formatting html style by `rehype-format`
5. Compile HAST(HTML Abstract Syntax Tree) to html string
6. Highlight html string, if there‘s any `<code>` tag

Now the result html look like this:
![image](https://github.com/remarkjs/remark/assets/62818957/0bc025e4-fc72-4746-8214-548b5b3fc8c6)
Every *keywords* is wrapped by a span with class named start with `hljs-`.

**Comment by ZacharyBear** [comment]
That's not enough, these classes are not styled.
Is there any highlight theme sets already exists, css sets, plugin or other tools...?

**Reply by wooorm** [reply]
yes, see the https://highlightjs.org website
